### PR TITLE
Create note listing page UI

### LINKED ISSUE #1: Create note listing page UI

### PR DESCRIPTION
**Pull request by vijay2136**
fixes #1 
- Implemented Sidebar
- Implemented Menubar
- Built list of notes page

### CODE REVIEW

File: app/javascript/src/components/Dashboard/Notes/index.jsx [hunk 1]
```diff
@@ -1,102 +1,60 @@
...
-          title="Looks like you don't have any notes!"
-          subtitle="Add your notes to send customized emails to them."
-          primaryAction={() => setShowNewNotePane(true)}
-          primaryActionLabel="Add New Note"
+    <>
+      <Menubar showMenu={showMenu} title="Notes" />
+      <Container>
+        <Header
+          title="All Notes"
+          menuBarToggle={() => setShowMenu(prev => !prev)}
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/index.jsx:25]: What's your reasoning for using prevState from within the setter over here rather than using `!showMenu`? What are the advantages of using this pattern over here?

File: app/javascript/src/components/Common/Menubar.jsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            <MenuBar.Block label="Users" count={80} />
+            <MenuBar.Block label="Leads" count={60} />
+            <MenuBar.Block label="Visitors" count={60} />
+          </>
+        )}
+        <MenuBar.SubTitle
+          iconProps={[
+            {
+              icon: () => <Search size={20} />,
+              onClick: () => setIsSearchCollapsed(!isSearchCollapsed),
```
Comment (yedhink) [inline app/javascript/src/components/Common/Menubar.jsx:33]: In reference to previous comment.

**Review by yedhink — COMMENTED** [review]
@vijay2136 

Few comments.

File: app/javascript/src/components/Dashboard/Notes/index.jsx (hunk 1 above)
Comment (vijay2136) [reply app/javascript/src/components/Dashboard/Notes/index.jsx:25]: I'm little confused here on what the best approach is to toggle state. Ive read somewhere that its best to use prevState  when toggling because the next state is dependent on the prevState. Can you please tell me what approach should i follow ?

Comment (yedhink) [reply app/javascript/src/components/Dashboard/Notes/index.jsx:25]: @vijay2136 

Go through the below mentioned articles and try to understand why we should be using previous state callback over here:
- https://stackoverflow.com/questions/58193166/usestate-hook-setter-incorrectly-overwrites-state
- https://medium.com/nerd-for-tech/react-hooks-with-closures-usestate-v-s-usereducer-9e0c20e81051
- https://dev.to/anpos231/react-hooks-the-closure-hell-71m

Once you go through them and understand what's going on, then please do type in a reply for this comment pointing in simple terms why we are preferring prevState over here. This is to ensure whether you've learn the right thing or not.

Comment (vijay2136) [reply app/javascript/src/components/Dashboard/Notes/index.jsx:25]: when you have a function defined inside of another function, that inner function has access to the variables and scope of the outer function, even if the outer function finishes executing and those variables are no longer accessible outside of that function. So its better to use prevState callback to avoid the closure problem.

Comment (yedhink) [reply app/javascript/src/components/Dashboard/Notes/index.jsx:25]: @vijay2136 

That's just a part of it.

Main issue is this: https://stackoverflow.com/a/58193404

Comment (vijay2136) [reply app/javascript/src/components/Dashboard/Notes/index.jsx:25]: Yes, I was trying to say that in simpler terms.
